Bluetooth: audio: ascs: Make stream->ep valid in bt_bap_stream_ops.re…

MariuszSkamra opened the pull request:
…leased

Once the application bt_bap_stream_ops.released callback is called make sure the stream->ep pointer is valid so that application can still access the endpoint details like e.g. endpoint direction.

Fixes: ASCS/SR/ACP/BV-{24,25,26,27,28,29,30,31}-C

MariuszSkamra committed:
Bluetooth: audio: ascs: Make stream->ep valid in bt_bap_stream_ops.released

Once the application bt_bap_stream_ops.released callback is called make
sure the stream->ep pointer is valid so that application can still access
the endpoint details like e.g. endpoint direction.

Fixes: ASCS/SR/ACP/BV-{24,25,26,27,28,29,30,31}-C
Signed-off-by: Mariusz Skamra <mariusz.skamra@codecoup.pl>

Thalley commented on subsys/bluetooth/audio/ascs.c:
```diff
@@ -247,11 +251,15 @@ static void ase_set_state_idle(struct bt_ascs_ase *ase)
 	ops = stream->ops;
 	if (ops != NULL && ops->released != NULL) {
 		ops->released(stream);
+		stream = NULL;
```
This operation does nothing

MariuszSkamra committed:
Bluetooth: audio: ascs: Add endpoint by stream lookup function

This adds endpoint by stream lookup function used to find the active
endpoints that use the stream object provided. The function is used
instead of dereferencing stream->ep that may be not valid if application
did not memset the stream object.

Signed-off-by: Mariusz Skamra <mariusz.skamra@codecoup.pl>
